src/python/pants/backend/codegen/protobuf/python/register.py

```diff
@@ -16,6 +17,7 @@ def rules():
         *additional_fields.rules(),
         *python_protobuf_subsystem.rules(),
         *python_rules(),
+        *write_codegen_goal.rules(),
```

Comment:
Right now, I'm only activating this new goal if you have `pants.backend.codegen.protobuf.python` activated, rather than unconditionally. But I'm not sure if that makes sense.

Where it matters is the `relocated_files` target. That uses codegen. Possibly, users want to be able to run `write-codegen` on that, even if they are not using Protobuf. On the other hand, it might be confusing to show that goal if you don't have Protobuf, as it's sort of an implementation detail that we're using codegen for it.

Idrk, any thoughts?

Reply:
Relocated files are a pretty arcane feature, and I'm not sure people will need to export that. I think it's fine to only enable this if a codegen backend is activated.

Reply:
Oh cc @kaos. I guess this makes sense. So I take back that I was going to have export-codegen always activated